napari/_qt/qt_main_window.py

```diff
@@ -874,21 +879,28 @@ def _add_viewer_dock_widget(
                     ]
                     if dock_widget.name in sub_actions:
                         idx = sub_actions.index(dock_widget.name)
-                        old_action = (
+                        current_action = (
                             self.plugins_menu.actions()[cnt]
                             .menu()
                             .actions()[idx]
                         )
-                        self.plugins_menu.actions()[cnt].menu().insertAction(
-                            old_action, action
-                        )
-                        self.plugins_menu.actions()[cnt].menu().removeAction(
-                            old_action
+                        dock_widget.closed.connect(
+                            lambda: current_action.setChecked(False)
                         )
+                        dock_widget.setVisible(True)
+
                         return
 
+        # it the action was not in the plugins menu, add it to the window menu.
         self.window_menu.addAction(action)
 
+    def _toggle_dock_visibility(self, dock_widget=None):
```

Comment:
can't see where this is used?

Reply:
it looks to be left over from an old idea. thanks for catching this.